code/datums/components/food/edible.dm

```diff
@@ -213,19 +211,26 @@ Behavior that's still missing from this component that original food items had t
 	if(eater.satiety > -200)
 		eater.satiety -= junkiness
 	playsound(eater.loc,'sound/items/eatfood.ogg', rand(10,50), TRUE)
-	if(owner.reagents.total_volume)
-		SEND_SIGNAL(parent, COMSIG_FOOD_EATEN, eater, feeder)
-		var/fraction = min(bite_consumption / owner.reagents.total_volume, 1)
-		owner.reagents.trans_to(eater, bite_consumption, transfered_by = feeder, methods = INGEST)
-		bitecount++
-		On_Consume(eater)
-		checkLiked(fraction, eater)
-
-		//Invoke our after eat callback if it is valid
-		if(after_eat)
-			after_eat.Invoke(eater, feeder)
-
-		return TRUE
+	var/atom/belly
+	if(istype(eater, /mob/living/carbon))
+		belly = eater.getorganslot(ORGAN_SLOT_STOMACH)
+	else
+		belly = eater
```

Comment:
```suggestion
	var/atom/belly = eater
	if(istype(eater, /mob/living/carbon))
		belly = eater.getorganslot(ORGAN_SLOT_STOMACH)
```
Might be worth making eater the default, up to you

Reply:
The reason to do this is to prevent eating with no stomach.
But i think the idea that anyone eating will be a carbon to start with and this can be cleaned up a bit.

Reply:
Ahhhh I see, didn't really think about that. Does getorganslot not return null on a failure?

Reply:
I failed to think of that